src/registry2.0/portimage.tcl

```diff
@@ -351,45 +351,45 @@ proc extract_archive_to_tmpdir {location} {
                 set tar "tar"
-                if {[catch {set tar [macports::findBinary $tar ${macports::autoconf::tar_path}]} errmsg] == 0} {
-                    ui_debug "Using $tar"
-                    set unarchive.cmd "$tar"
-                    set unarchive.pre_args {-xvpf}
-                    if {[regexp {z2?$} ${unarchive.type}]} {
-                        set unarchive.args {-}
-                        if {[regexp {bz2?$} ${unarchive.type}]} {
-                            if {![catch {macports::binaryInPath lbzip2}]} {
-                                set gzip "lbzip2"
-                            } elseif {![catch {macports::binaryInPath pbzip2}]} {
-                                set gzip "pbzip2"
-                            } else {
-                                set gzip "bzip2"
-                            }
-                        } elseif {[regexp {lz$} ${unarchive.type}]} {
-                            set gzip "lzma"
-                        } elseif {[regexp {xz$} ${unarchive.type}]} {
-                            set gzip "xz"
-                        } else {
-                            set gzip "gzip"
-                        }
-                        if {[info exists macports::autoconf::${gzip}_path]} {
-                            set hint [set macports::autoconf::${gzip}_path]
+                if {[catch {set tar [macports::findBinary $tar ${macports::autoconf::tar_path}]} errmsg]} {
+                    ui_debug $errmsg
+                    throw MACPORTS "No '$tar' was found on this system!"
+                }
+
+                ui_debug "Using $tar"
+                set unarchive.cmd "$tar"
+                set unarchive.pre_args {-xvpf}
+                if {[regexp {z2?$} ${unarchive.type}]} {
+                    set unarchive.args {-}
+                    if {[regexp {bz2?$} ${unarchive.type}]} {
+                        if {![catch {macports::binaryInPath lbzip2}]} {
+                            set gzip "lbzip2"
+                        } elseif {![catch {macports::binaryInPath pbzip2}]} {
+                            set gzip "pbzip2"
                         } else {
-                            set hint ""
-                        }
-                        if {[catch {set gzip [macports::findBinary $gzip $hint]} errmsg] == 0} {
-                            ui_debug "Using $gzip"
-                            set unarchive.pipe_cmd "$gzip -d -c ${location} |"
-                        } else {
-                            ui_debug $errmsg
-                            throw MACPORTS "No '$gzip' was found on this system!"
+                            set gzip "bzip2"
                         }
+                    } elseif {[regexp {lz$} ${unarchive.type}]} {
+                        set gzip "lzma"
+                    } elseif {[regexp {xz$} ${unarchive.type}]} {
+                        set gzip "xz"
+                    } else {
+                        set gzip "gzip"
```

Comment:
While we're changing this anyway, should we run `s/gzip/compression/g`? `set gzip "xz"` just looks weird to me.